tests/conformance/bindings/bindings.go

```diff
@@ -36,44 +39,67 @@ func (tc *TestConfig) createInvokeRequest() bindings.InvokeRequest {
 	}
 }
 
-func ConformanceTests(t *testing.T, props map[string]string, binding bindings.OutputBinding, config TestConfig) {
+func ConformanceTests(t *testing.T, props map[string]string, inputBinding bindings.InputBinding, outputBinding bindings.OutputBinding, config TestConfig) {
+	// We need to increase the number of processes as the input binding is its own process that kicks off multiple sub processes
+	runtime.GOMAXPROCS(20)
+
 	// Init
 	t.Run("init", func(t *testing.T) {
-		err := binding.Init(bindings.Metadata{
-			Properties: props,
-		})
-		assert.NoError(t, err, "expected no error setting up binding")
+		// Check for an output binding specific operation before init
+		if config.HasOperation("operations") {
+			err := outputBinding.Init(bindings.Metadata{
+				Properties: props,
+			})
+			assert.NoError(t, err, "expected no error setting up output binding")
+		}
+		// Check for an input binding specific operation before init
+		if config.HasOperation("read") {
+			err := inputBinding.Init(bindings.Metadata{
+				Properties: props,
+			})
+			assert.NoError(t, err, "expected no error setting up input binding")
+		}
 	})
 
 	// Operations
 	if config.HasOperation("operations") {
 		t.Run("operations", func(t *testing.T) {
-			ops := binding.Operations()
+			ops := outputBinding.Operations()
 			for _, op := range ops {
 				assert.True(t, config.HasOperation(string(op)), fmt.Sprintf("Operation missing from conformance test config: %v", op))
 			}
 		})
 	}
 
-	// CreateOperations
+	inputBindingCall := 0
+	var wg sync.WaitGroup
+	if config.CommonConfig.HasOperation("read") {
```

Comment:
```suggestion
	if config.HasOperation("read") {
```

can remove CommonConfig

Reply:
Done.